### PR TITLE
Reduce chat spam

### PR DESCRIPTION
**Pull request by Tayyyyyyy**
A bit of javascript to condense chat lines if the same one happens a bunch of times, by default set to on. Not sure if I like the appearance of the badge. Would like input on what people would want this to look like:

### Gif:
gif

### Before:
before_badge

### After:
after_badge

### Preferences:
badge_preferences

:cl: Tayyyyyyy
add: Repeated lines are now combined in the chat window. This can be disabled in chat window preferences.
/:cl:

### CODE REVIEW

**Comment by tigercat2000** [comment]
This should be a user setting, not automatic.

**Comment by Tayyyyyyy** [comment]
Kay, lemme add that in...

**Comment by KasparoVy** [comment]
Goon-style! Really nice.

**Comment by Tayyyyyyy** [comment]
> Goon-style! Really nice.

I knew Goon had this, but for the life of me couldn't find where it was implemented in their code...

@tigercat2000 Preferences added (and main post updated with a screenshot)! It's set to off by default and uses the cookie to save your preference.

**Comment by deep-fried-faceplant** [comment]
The chat spam condenser isn't a part of the 2016 open release, goon added it a month or two ago.

**Comment by Dumbdumn5** [comment]
:+1:

**Comment by Tayyyyyyy** [comment]
Okay, Lawbringer brought up with me in OOC that if it's off by default, people may not realize it was added, so I've made it on by default. If there are any objections to this, I can revert this commit.

**Comment by SamHPurp** [comment]
@Tayyyyyyy makes sense. I'd reword the changelog though, maybe "Repeated lines are now combined in the chat window. This can be disabled in preferences."

**Comment by Spacemanspark** [comment]
Yes please.

**Comment by Tayyyyyyy** [comment]
Added a small animation to the badge popping in, since all the cool kids at oracle have it 🙂 hold on while I try to figure out how to record a gif...

**Comment by SamHPurp** [comment]
>Added a small animation to the badge popping in, since all the cool kids at oracle have it

o7

My only concern on this is encouraging spam. If there was a way to exclude this from player inputted messages, it might be a good idea.

**Comment by tigercat2000** [comment]
:+1: :x: :100: with this being a user preference. Very nice work @Tayyyyyyy

**Comment by Tayyyyyyy** [comment]
@SamHPurp I think that could be easily done by looking at the css class of the previous message, but hopefully the badge animation isn't so crazy that people will want to spam ooc for it. Making the text bigger, on the other hand...

**Comment by SamHPurp** [comment]
Yep, I agree with you -  your solution with the badge is much less encouraging of spam.
